public/static/js/settings.js

```diff
@@ -80,10 +81,49 @@ mciModule.controller('SettingsCtrl', ['$scope', '$http', '$window', 'notificatio
       });
   }
 
-  $scope.updateUserSettings = function(new_tz, use_spruce_options) {
+  $scope.showAdditionalFields = function(){
+    var additionalFieldCount = 1;
+    var additionalField = document.getElementById("feedback-" + additionalFieldCount);
+
+    while (additionalField) {
+      if (use_spruce_options.patch_page) {
+        additionalField.style.display = "block";
+      } else {
+        additionalField.style.display = "none";
+      }
+      additionalFieldCount++;
+      additionalField = document.getElementById("feedback-" + additionalFieldCount);
+    }
+  }
+
+  function formatFeedback(spruce_feedback) {
+    var formattedFeedback = { type: "new_patches_page_feedback" };
+    var questionAnswerArray = [];
+    var allFields = ["information_score", "usability_score", "missing_things", "requested_changes"];
+    allFields.forEach( function(field) {
+        var prompt = document.getElementById(field + "_prompt").innerHTML;
+        var answer = spruce_feedback[field] === undefined ? "" : spruce_feedback[field]
+        var questionAnswer = {
+          id: field,
+          prompt: prompt,
+          answer: answer
+        };
+        questionAnswerArray.push(questionAnswer);
+    });
+    formattedFeedback["questions"] = questionAnswerArray;
+    return formattedFeedback;
+  }
+
+  $scope.updateUserSettings = function(new_tz, use_spruce_options, spruce_feedback) {
+    if (use_spruce_options.patch_page && document.getElementById("feedback-1").style.display === "block" &&
```

Comment:
Instead if checking if the field is visible, check the state of $scope.use_spruce_options.patch_page. This will separate logic from UI